seg_lapa/train.py

```diff
@@ -54,21 +54,14 @@ def main(cfg: TrainConfig):
     data_conf = instantiate(cfg.dataset)
     dm = data_conf.get_datamodule()
     print(dm)
-    exit()
-
-    trainer = pl.Trainer(gpus=[0], overfit_batches=0.0,
-                         distributed_backend="ddp", num_nodes=1,
-                         precision=32,
-                         limit_train_batches=1.0,
-                         limit_val_batches=1.0,
-                         limit_test_batches=1.0,
-                         max_steps=cfg.num_steps,
-                         fast_dev_run=False,
-                         )
+
+    trainer_conf = instantiate(cfg.trainer)
+    trainer = trainer_conf.get_trainer()
+
```

Comment:
```suggestion
    config = train_conf.parse_config(cfg)
    trainer = config.trainer.get_trainer()
    dm = config.dataset.get_datamodule()
```

Apologies for the confusion. We don't want to use hydra's instantiate functionality anywhere. I don't like how it works, it's too opaque. We created our config parsing class to replace it, usage demonstrated above.

We should remove the instantiate function used above these lines for the datamodule as well.

We also need to modify `class DeeplabV3plus` so that it accepts the config object and can use it to instantiate the optimizer.
```
class DeeplabV3plus(pl.LightningModule):

    def __init__(self, config: TrainConf):
        super().__init__()
        self.model = DeepLab(backbone='drn', output_stride=8, num_classes=11,
                             sync_bn=False, enable_amp=False)
        self.config = config
        ...
        
    def configure_optimizers(self):
        optimizer = config.optimizer.get_optimizer(self.parameters())
        return optimizer
```